File: app-java/src/main/java/science/monke/api/order/entity/OrderResponse.java
package science.monke.api.order.entity;

import lombok.Builder;
import science.monke.internal.order.entity.OrderId;

import java.time.OffsetDateTime;
import java.util.Set;

public record OrderResponse(
        OrderStatus status,
        OrderId orderId,
        Set<OrderItemResponse> items,
        OffsetDateTime creationDate
) {
    // Can be moved to class level if IntelliJ plugin works
    @Builder
    public OrderResponse {}
}

File: app-kotlin/src/main/kotlin/science/monke/api/order/entity/OrderResponse.kt
package science.monke.api.order.entity

import science.monke.internal.order.entity.OrderId
import java.time.OffsetDateTime

data class OrderResponse(
    val status: OrderStatus,
    val orderId: OrderId,
    val items: Set<OrderItemResponse>,
    val creationDate: OffsetDateTime
)
